### PR TITLE
Use secure random generator instead of pseudo-random generator

### PR DESCRIPTION
**Pull request by thegoncalomartins**
## What does this PR do?

This PR replaces the usage of pseudo-random number generator `Math.random()` with the secure random number generator `crypto.getRandomValues()` or `crypto.randomInt()`.
The motivation for this is because my team at [Talkdesk](https://github.com/Talkdesk) built a library that uses the `pusher-js` as a dependency and our library was analyzed by a SAST tool that alerted that using the simple `Math.random()` generator is not secure enough.

## Checklist

- [X] All new functionality has tests.
- [X] All tests are passing.
- [X] New or changed API methods have been documented.
- [X] `npm run format` has been run

### CODE REVIEW

File: types/src/core/strategies/transport_strategy.d.ts [hunk 1]
```diff
@@ -9,6 +9,9 @@ export default class TransportStrategy implements Strategy {
     constructor(name: string, priority: number, transport: Transport, options: StrategyOptions);
     isSupported(): boolean;
     connect(minPriority: number, callback: Function): {
+        abort: () => void;
```
Comment (limadelrey) [inline types/src/core/strategies/transport_strategy.d.ts:12]: Why is this needed? 💭

File: src/runtimes/node/runtime.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { Network } from './net_info';
 import xhrAuth from 'isomorphic/auth/xhr_auth';
 import { AuthTransports } from 'core/auth/auth_transports';
 import xhrTimeline from 'isomorphic/timeline/xhr_timeline';
+import * as crypto from 'crypto';
```
Comment (limadelrey) [inline src/runtimes/node/runtime.ts:9]: Could you only import the function we need (`randomInt`)?

File: src/runtimes/react-native/runtime.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { Network } from './net_info';
 import xhrAuth from 'isomorphic/auth/xhr_auth';
 import { AuthTransports } from 'core/auth/auth_transports';
 import xhrTimeline from 'isomorphic/timeline/xhr_timeline';
+import * as crypto from 'crypto';
```
Comment (limadelrey) [inline src/runtimes/react-native/runtime.ts:7]: Same as before ☝️

**Review by limadelrey — COMMENTED** [review]
I've left some minor suggestions, but overall it looks good to me 👍

File: types/src/core/strategies/transport_strategy.d.ts (hunk 1 above)
Comment (thegoncalomartins) [reply types/src/core/strategies/transport_strategy.d.ts:12]: I didn't add this. I think it was added automatically after I ran the commands to format the code and/or running the unit and integration tests.

File: src/runtimes/node/runtime.ts (hunk 2 above)
Comment (thegoncalomartins) [reply src/runtimes/node/runtime.ts]: Done.

File: src/runtimes/react-native/runtime.ts (hunk 3 above)
Comment (thegoncalomartins) [reply src/runtimes/react-native/runtime.ts]: Done.

**Review by limadelrey — APPROVED** [review]

**Comment by sashaafm** [comment]
I concur this change would be beneficial to us 🙏

**Review by sashaafm — APPROVED** [review]
LGTM we'd love to see this merged into `master`

**Comment by amad** [comment]
@thegoncalomartins I liked your previous implementation using `crypto.randomInt`. I might be wrong, but I think updating `@types/node` could solve that error.

**Comment by thegoncalomartins** [comment]
@amad Can you please approve again running the workflow?
When can we have feedback on this or have the PR merged? It's impacting our work and we have a deadline to solve the problem...
cc @sashaafm @limadelrey @armandovnobre

**Comment by fbenevides** [comment]
@thegoncalomartins tests are still broken. A solution for that would be to bump `@types/node` version using:
```
npm install @types/node latest
```

**Comment by fbenevides** [comment]
Hi @thegoncalomartins could you please sync your repo with the newest version of our master/main branch? We've bumped the `@types/node` version, which will fix your PR.

**Comment by thegoncalomartins** [comment]
> Hi @thegoncalomartins could you please sync your repo with the newest version of our master/main branch? We've bumped the `@types/node` version, which will fix your PR.

@fbenevides I just synced my repo with the newest version. Thanks 🙏

**Review by fbenevides — APPROVED** [review]
